### ISSUE TITLE
No error for accessing base class properties before super call if derived class has no parameter properties or initialized properties

### ISSUE THREAD

**Issue by JsonFreeman** [Microsoft/TypeScript#3876]
When a derived class has parameter properties or initialized properties, we require

* the super call to be first
* the super call not to reference this

That's because the derived class's properties will be initialized after the super call. But we also know that base class properties will be initialized *inside* the super call. Therefore, it would seem that a base property access should be prohibited before the super call, and in the super call statement itself. We get no error for these, but arguably we should.
```ts
class Base {
    y = 0;
    constructor(public x, y) { }
}

class Derived extends Base {
    constructor() {
        this.x; // No error for these two
        this.y;
        super(this.x, this.y); // No error for these either
    }
}
```

**Comment by yuit** [comment]
```typescript
class A { }
class B extends A {
    public x: number = 20;
    constructor() {
        this.x = 10;
        super();
    }
}
```
Extra node: above code will give an error

**Comment by JsonFreeman** [comment]
Yes because B has an initialized property. But if A had the initialized property instead, we would not give the error.

**Comment by mhegazy** [comment]
this should be flagged as an error now.
